hackathon_site/registration/admin.py

```diff
@@ -1,13 +1,65 @@
+import csv
+import functools
+
 from django.contrib import admin
+from django.http import HttpResponse
+
 from registration.models import Application, Team as TeamApplied
 
 
+def rgetattr(obj, attr, *args):
+    """
+    Recursive getattr, allows for nested attributes.
+    For example if you're in the Application class, you can't do getattr for user.first_name
+    But with rgetattr you can.
+    """
+
+    def _getattr(obj, attr):
+        return getattr(obj, attr, *args)
+
+    return functools.reduce(_getattr, [obj] + attr.split("."))
+
+
 class ApplicationInline(admin.TabularInline):
     model = Application
     autocomplete_fields = ("user",)
     extra = 0
 
 
+class ExportCsvMixin:
+    export_fields = []
+
+    def export_as_csv(self, request, queryset):
+        # Setup export_field_names and export_field attributes arrays
+        export_field_names = []
+        export_field_attributes = []
+        if self.export_fields:
+            for item in self.export_fields:
+                export_field_names.append(item[0])
+                export_field_attributes.append(item[1])
+
+        response = HttpResponse(content_type="text/csv")
+        response["Content-Disposition"] = "attachment; filename={}.csv".format(
+            self.model._meta
+        )
+        writer = csv.writer(response)
+
+        writer.writerow(export_field_names)
+        for obj in queryset:
+            attributes = []
+            for field in export_field_attributes:
+                try:
+                    attributes.append(rgetattr(obj, field))
+                except Application.review.RelatedObjectDoesNotExist:
```

Comment:
I think it would be better to except `django.core.exceptions.ObjectDoesNotExist`, that way it's resilient to other fields being not found as well if we use this mixin other places